src/tools/android/java/com/google/devtools/build/android/ResourceProcessorBusyBox.java

```diff
@@ -183,36 +184,43 @@ private static int runPersistentWorker() throws Exception {
     PrintStream ps = new PrintStream(buf, true);
     PrintStream realStdOut = System.out;
     PrintStream realStdErr = System.err;
-    try {
-      // Redirect all stdout and stderr output for logging.
-      System.setOut(ps);
-      System.setErr(ps);
-
-      while (true) {
-        try {
-          WorkRequest request = WorkRequest.parseDelimitedFrom(System.in);
-          if (request == null) {
-            break;
-          }
-
-          int exitCode = processRequest(request.getArgumentsList());
-          ps.flush();
-
-          WorkResponse.newBuilder()
-              .setExitCode(exitCode)
-              .setRequestId(request.getRequestId())
-              .setOutput(buf.toString())
-              .build()
-              .writeDelimitedTo(realStdOut);
 
-          realStdOut.flush();
-          buf.reset();
-        } catch (IOException e) {
-          logger.severe(e.getMessage());
-          e.printStackTrace(realStdErr);
-          return 1;
-        }
-      }
+    // Redirect all stdout and stderr output for logging.
+    System.setOut(ps);
+    System.setErr(ps);
+    try {
+      WorkRequestHandler workerHandler =
+          new WorkRequestHandler.WorkRequestHandlerBuilder(
+                  new WorkRequestHandler.WorkRequestCallback(
+                      (request, pw) -> {
+                        try {
+                          processRequest(request.getArgumentsList());
+                          return 0;
+                        } catch (Exception e) {
+                          e.printStackTrace(pw);
+                          return 1;
+                        } finally {
+                          try {
+                            buf.flush();
+                            String s = buf.toString().trim();
+                            buf.reset();
+                            if (!s.isEmpty()) {
+                              pw.print(s);
```

Comment:
This sends everything sent to stdout/err into the `output` field of the response. However, since System.out/err are global, output from all threads gets interleaved here. These should be sent to `realStdErr`, any actual error messages from processing this request should go to `pw`. See discussion in https://github.com/bazelbuild/bazel/pull/14201

Reply:
Merging https://github.com/bazelbuild/bazel/pull/14201 first would be the preferred solution here so that we aren't having to duplicate any of the system stream remapping logic between workers.

Reply:
Pushed the changes to swap `System.err` in for `System.out`.

Reply:
Also worth noting that swapping `System.err` in for `System.out` technically changes where the outputs are delivered for this rule, where the prior implementation would write the captured outputs directly to the work response output vs. the new implementation dumps them to `System.err` and only writes the contents of the `PrintWriter` provided by `WorkRequestHandler` to the work response output.

Reply:
Then you need to be sure that that's the desired behaviour. Things the user should see should go into the work response output, things that should go into the log should go to `System.out`.

Reply:
I'm going to roll back to the previous behavior to preserve the existing logs/error delivery that exists today.